flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/util/serialization/JSONKeyValueDeserializationSchema.java

```diff
@@ -37,6 +38,7 @@
  * <p>Metadata fields can be accessed by calling objectNode.get("metadata").get(&lt;name>).as(&lt;type>) and include
  * the "offset" (long), "topic" (String) and "partition" (int).
  */
+@Internal
```

Comment:
This could also be `@PublicEvolving`, I don't think there's a problem with the user using it directly.